xtask/rules_check/src/lib.rs

```diff
@@ -405,10 +370,8 @@ fn assert_lint(
                 });
             }
         }
-        // Foreign code blocks should be already ignored by tests
-        BlockType::Foreign(block) => {
-            bail!("Unrecognised block type {}", &block)
-        }
+        // Unknown code blocks should be already ignored by tests
+        DocumentFileSource::Unknown => {}
```

Comment:
We should be more forgiving here, in case we want to add some foreign code blocks (like `shell`) for illustration purpose.